go/dpagg/sum_test.go

```diff
@@ -149,9 +168,28 @@ func TestBoundedSumFloat64Serialization(t *testing.T) {
 		if !cmp.Equal(bsUnchanged, bsUnmarshalled, cmp.Comparer(compareBoundedSumFloat64)) {
 			t.Errorf("decode(encode(_)): when %s got %v, want %v", tc.desc, bsUnmarshalled, bs)
 		}
-		// Check that the original BoundedSumFloat64 has its resultReturned set to true after serialization.
-		if !bs.resultReturned {
-			t.Errorf("BoundedSumFloat64 %v should have its resultReturned set to true after being serialized", bs)
+		if bs.state != Serialized {
+			t.Errorf("BoundedSumFloat64 should have its state set to Serialized, got %v, want Serialized", bs.state)
+		}
+	}
+}
+
+// Tests that GobEncode() returns errors correctly with different BoundedSumFloat64 aggregation states.
+func TestBoundedSumFloat64SerializationStateChecks(t *testing.T) {
+	for _, tc := range []struct {
+		state   aggregationState
+		wantErr bool
+	}{
+		{Default, false},
+		{Merged, true},
+		{Serialized, true},
+		{ResultReturned, true},
+	} {
+		bs := getNoiselessBSF()
+		bs.state = tc.state
+
+		if _, err := bs.GobEncode(); (err != nil) != tc.wantErr {
+			t.Errorf("GobEncode: when state %v for err got %v, want %t", tc.state, err, tc.wantErr)
```

Comment:
nit: "wantErr" instead of "want" inside the string. (Could you also check the other aggregations for the same fix?)

Reply:
I have added that. I have also added it to the old merge compatibility tests since they had the same message.
https://github.com/google/differential-privacy/pull/57/files#diff-76b5b3637fcd413f3c05f48243f09d20R298-R299